craft_parts/executor/executor.py

```diff
@@ -91,7 +91,7 @@ def prologue(self) -> None:
 
         # update the overlay environment package list to allow installation of
         # overlay packages.
-        if any(p.has_overlay for p in self._part_list):
```

Comment:
if p.has_overlay is unused, it might be worth removing

Reply:
it does have other uses (unrelated to packages)